src/model/Publication.ts

```diff
@@ -244,7 +236,8 @@ export default class Publication {
   /**
    * positionsByHref
    */
-  public positionsByHref(href: string) {
-    return this.positions.filter((el: Locator) => el.href === decodeURI(href));
+  public positionsByHref(href: string): Locator[] {
+    const decodedHref = decodeURI(href) ?? "";
+    return this.positions.filter((p: Locator) => decodedHref.includes(p.href));
```

Comment:
Rather than checking if the decodedhref is strictly equal to the `el.href` we can just compare the two in a more dynamic way using `String.prototype.includes`

Reply:
Do you think that this change could be generally useful to R2 users?  Meaning, could this be a pull request to R2 repo that everyone could use?

Reply:
Yes I do think so. The change here removes the requirement that manifestUrl match a specific pattern, so the code is more flexible with this change. It caused us a lot of problems!